### PR TITLE
[refactor] moving memtracer to gemini

### PR DESCRIPTION
**Pull request by feifeibear**

### CODE REVIEW

**Comment by feifeibear** [comment]
image

File: colossalai/engine/ophooks/_memtracer_ophook.py [hunk 1]
```diff
@@ -25,6 +23,7 @@ class MemTracerOpHook(BaseOpHook):
     """
 
     def __init__(self, warmup: int = 50, refreshrate: int = 10, data_prefix: str = "memstats"):
+        from colossalai.gemini.memory_tracer import AsyncMemoryMonitor
```
Comment (ver217) [inline colossalai/engine/ophooks/_memtracer_ophook.py:26]: Why import in function?

Comment (feifeibear) [reply colossalai/engine/ophooks/_memtracer_ophook.py:26]: circle import. hard to fix.

Comment (feifeibear) [reply colossalai/engine/ophooks/_memtracer_ophook.py:26]: circle import.
We will fix the import in the next PR

**Review by ver217 — APPROVED** [review]

**Pull request merged**